kotlin/samples/recyclerview/src/main/java/com/squareup/sample/recyclerview/ListDiffMode.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.recyclerview
+
+import com.squareup.workflow.ui.ContainerHintKey
+import kotlinx.coroutines.CoroutineDispatcher
+
+sealed class ListDiffMode {
+
+  object None : ListDiffMode()
+  object Synchronous : ListDiffMode()
+  data class Asynchronous(val mainDispatcher: CoroutineDispatcher) : ListDiffMode()
```

Comment:
I guess we're using it for DI… :D

Reply:
I was going to add a 4th recycler that used `Main.immediate`… I should probably remove this.

Reply:
Removed.